request.go

```diff
@@ -100,6 +101,12 @@ func NewSerializer(opts RequestOptions) (encoding.Serializer, error) {
 	e := detectEncoding(opts)
 	if e == encoding.Thrift {
 		return encoding.NewThrift(opts.ThriftFile, opts.Procedure, opts.ThriftMultiplexed)
+	} else if e == encoding.Protobuf {
```

Comment:
can you change this code to use a `switch` instead since there's multiple cases now,
```
switch e := detectEncoding(opts); e {
case encoding.Thrift: 
  [...]
case encoding.Protobuf:
  [...]
}
```

Reply:
Moved this down a few lines now. Protobuf currently doesn't support the "if no method is specified return the list of available methods" behaviour. Leaving thrift special cased here and moving the proto encoding to the already existing switch statement for json and raw encodings.

Reply:
Does protobuf not support in our current implementation, or does the underlying protoreflect package not support it? If it's just our implementation, then we should implement it as a follow-up, and structure the code assuming we will and leave a TODO. I found this method which makes it seems like it should be possible using protoreflect:
https://godoc.org/github.com/jhump/protoreflect/desc#ServiceDescriptor.GetMethods

Reply:
It should support it. Let me add a todo for it.